realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -435,6 +435,9 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_SharedRealm_nativeGetTable(JNIEnv
         auto& shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
         if (!shared_realm->read_group().has_table(name)) {
             std::string name_str = name;
+            if (name_str.find(TABLE_PREFIX) == 0) {
```

Comment:
Java should always pass a `table_name` here which means it should always have `TABLE_PREFIX` in the string. What is the case triggered this issue?

(Due to historical reasons, we sometimes mixed use of table_name and class_name. Ideally the `TABLE_PREFIX` should be hidden from java -- java doesn't have to know that. By integrating more object_store stuff, we are slowly moving to the target.)